FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m

```diff
@@ -18,56 +18,6 @@
 
 @implementation FIRAppDistributionKeychainUtility
 
-+ (void)handleAuthStateError:(NSError **_Nullable)error
-                 description:(NSString *)description
-                        code:(int)code {
-  if (error) {
-    NSDictionary *userInfo = @{NSLocalizedDescriptionKey : description};
-    *error = [NSError errorWithDomain:kFIRAppDistributionKeychainErrorDomain
-                                 code:code
-                             userInfo:userInfo];
-  }
-}
-
-+ (BOOL)addKeychainItem:(nonnull NSMutableDictionary *)keychainQuery
-     withDataDictionary:(nonnull NSData *)data {
-  [keychainQuery setObject:data forKey:(id)kSecValueData];
-  OSStatus status = SecItemAdd((CFDictionaryRef)keychainQuery, NULL);
-
-  return status == noErr ? YES : NO;
-}
-
-+ (BOOL)updateKeychainItem:(nonnull NSMutableDictionary *)keychainQuery
-        withDataDictionary:(nonnull NSData *)data {
-  OSStatus status =
-      SecItemUpdate((CFDictionaryRef)keychainQuery, (CFDictionaryRef) @{(id)kSecValueData : data});
-  return status == noErr ? YES : NO;
-}
-
-+ (BOOL)deleteKeychainItem:(nonnull NSMutableDictionary *)keychainQuery {
-  OSStatus status = SecItemDelete((CFDictionaryRef)keychainQuery);
-
-  return status != errSecSuccess && status != errSecItemNotFound ? NO : YES;
-}
-
-+ (NSData *)fetchKeychainItemMatching:(nonnull NSMutableDictionary *)keychainQuery
-                                error:(NSError **_Nullable)error {
-  NSData *keychainItem;
-  OSStatus status = SecItemCopyMatching((CFDictionaryRef)keychainQuery, (void *)&keychainItem);
-
-  if (status != noErr || 0 == [keychainItem length]) {
-    if (error) {
-      NSString *description =
-          NSLocalizedString(@"Failed to fetch keychain item.",
-                            @"Error message for failure to retrieve auth state from keychain");
-      [self handleAuthStateError:error description:description code:0];
-      return nil;
-    }
-  }
-
-  return keychainItem;
-}
-
 + (OIDAuthState *)unarchiveKeychainResult:(NSData *)result {
```

Comment:
I considered it. But I decided to keep these 2 methods for testing.

Reply:
You should be able to mock these on the auth keychain persistence itself rather than in a separate class, but if this proves too difficult, then I'm cool with this.

Reply:
Done!